packages/test-runner/src/fixtures.ts

```diff
@@ -184,16 +201,20 @@ export class FixturePool {
 }
 
 export function fixturesForCallback(callback: Function): string[] {
-  const names = new Set<string>();
-  const visit  = (callback: Function) => {
+  const names = new Array<string>();
+  const name2Idx = new Map<string, number>();
+  const visit = (callback: Function) => {
     for (const name of fixtureParameterNames(callback)) {
-      if (name in names)
-        continue;
-      names.add(name);
+      names.push(name);
       if (!registrations.has(name))
         throw new Error('Using undefined fixture ' + name);
-
-      const { fn } = registrations.get(name);
+      if (!name2Idx.has(name))
+        name2Idx.set(name, 0);
+      else
+        name2Idx.set(name, name2Idx.get(name) + 1);
```

Comment:
This logic is pretty similar to the logic which we have in line 86. Thats why I thought about abstraction.